DBM-Core/DBM-Nameplate.lua

```diff
@@ -349,6 +412,336 @@ function nameplateFrame:Hide(isGUID, unit, spellId, texture, force, isHostile, i
+					aura_tbl.pauseStartTime = aura_tbl.startTime
+
+					NameplateIcon_UpdateUnitAuras(true,guid)
+				elseif not aura_tbl.paused and guid then
+					aura_tbl.paused = true
+					aura_tbl.pauseStartTime = GetTime()
+
+					NameplateIcon_UpdateUnitAuras(true,guid)
+				end
+			end
+		end
+	end
+end
+
+--------------------------------
+--  Nameplate Icon Functions  --
+--------------------------------
+--/run DBM:FireEvent("BossMod_EnableHostileNameplates")
+--/run DBM.Nameplate:Show(true, UnitGUID("target"), 227723)--Mana tracking, easy to find in Dalaran
+--/run DBM.Nameplate:Show(false, "target", 227723)--Mana tracking, easy to find in Dalaran
+--/run DBM.Nameplate:Hide(true, nil, nil, nil, true)
+--/run DBM.Nameplate:Hide(true, UnitGUID("target"), 227723)
+--/run DBM.Nameplate:Hide(false, GetUnitName("target", true), 227723)
+
+--isGUID: guid or name (bool)
+--ie, anchored to UIParent Center (ie player is in center) and to bottom of nameplate aura.
+function nameplateFrame:Show(isGUID, unit, spellId, texture, duration, desaturate, forceDBM)
+
+	-- nameplate icons are disabled;
+	if DBM.Options.DontShowNameplateIcons then return end
+
+	-- ignore player nameplate;
+	if playerGUID == unit or playerName == unit then return end
+
+	--Texture Id passed as string so as not to get confused with spellID for GetSpellTexture
+	local currentTexture = tonumber(texture) or texture or GetSpellTexture(spellId)
+
+	-- build aura info table
+	local curTime = GetTime()
+	local aura_tbl = {
+		texture = currentTexture,
+		spellId = spellId,
+		duration = duration or 0,
+		desaturate = desaturate or false,
+		startTime = curTime,
+		auraType = 1, -- 1 = nameplate aura; 2 = nameplate CD timers
+		index = currentTexture,
+	}
+
+	-- Supported by nameplate mod, passing to their handler
+	if SupportedNPModIcons() and not forceDBM then
+		DBM:FireEvent("BossMod_EnableHostileNameplates") --TODO: is this needed?
+		DBM:FireEvent("BossMod_ShowNameplateAura", isGUID, unit, aura_tbl.texture, aura_tbl.duration, aura_tbl.desaturate)
+		DBM:Debug("DBM.Nameplate Found supported NP mod, only sending Show callbacks", 3)
+		return
+	end
+
+	--call internal show function
+	NameplateIcon_Show(isGUID, unit, aura_tbl)
+end
+
+function nameplateFrame:Hide(isGUID, unit, spellId, texture, force, isHostile, isFriendly, forceDBM)
+	--Texture Id passed as string so as not to get confused with spellID for GetSpellTexture
+	local currentTexture = tonumber(texture) or texture or GetSpellTexture(spellId)
+
+	if SupportedNPModIcons() and not forceDBM then --aura icon handling
+		--Not running supported NP Mod, internal handling
+		DBM:Debug("DBM.Nameplate Found supported NP mod, only sending Hide callbacks", 3)
+
+		if force then
+			if isFriendly then
+				DBM:FireEvent("BossMod_DisableFriendlyNameplates")
+			end
+			if isHostile then
+				DBM:FireEvent("BossMod_DisableHostileNameplates")
+			end
+		elseif unit then
+			DBM:FireEvent("BossMod_HideNameplateAura", isGUID, unit, texture)
```

Comment:
This should pass `currentTexture` instead of `texture`

Reply:
Good catch, thank you. Fixed. :)